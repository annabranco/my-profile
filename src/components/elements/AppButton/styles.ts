import styled, { css } from 'styled-components';
import { rgba } from 'polished';
import {
  colorWhite,
  colorBlueLight,
  colorBlack,
  fontTitle
} from '../../../styles/theme';

interface StyledButtonProps extends React.ComponentPropsWithoutRef<'button'> {
  buttonStyle: string;
}

export const StyledButton = styled.button<StyledButtonProps>`
  border: 1px solid ${colorBlack};
  border-radius: 15px 0;
  margin: 5px auto;
  background-image: linear-gradient(
    ${rgba(colorWhite, 0.2)},
    ${rgba(colorBlueLight, 0.6)}
  );
  width: 100px;
  padding: 5px 0;
  text-decoration: none;
  font-family: ${fontTitle};
  font-size: 1.2rem;
  cursor: pointer;
  box-shadow: 0 2px 4px 0 rgba($black, 0.4),
    inset 0 1px 5px 1px rgba($white, 0.7);

  &:hover {
    transform: translate(0, 1px);
    text-decoration: none;
    box-shadow: inset 0 2px 5px 1px rgba($black, 0.4);
  }

  ${({ buttonStyle }) => {
    return (
      buttonStyle &&
      css`
        color: ${buttonStyle};
      `
    );
  }}
`;
StyledButton.displayName = 'App Button';
