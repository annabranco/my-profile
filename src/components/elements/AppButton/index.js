import React from 'react';
import { func, string } from 'prop-types';
import { StyledButton } from './styles';

const AppButton = ({ buttonStyle, onClick, text }) => (
  <StyledButton buttonStyle={buttonStyle} onClick={onClick} type="button">
    {text}
  </StyledButton>
);

AppButton.propTypes = {
  buttonStyle: string,
  onClick: func,
  text: string.isRequired
};

AppButton.defaultProps = {
  buttonStyle: 'default',
  onClick: () => null
};

export default AppButton;
