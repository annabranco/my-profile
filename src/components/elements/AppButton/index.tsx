import { ReactElement } from 'react';
import * as React from 'react';
import { func, string } from 'prop-types';
import { StyledButton } from './styles';

interface Props {
  buttonStyle: string;
  datae2eid: string;
  onClick: (event: React.MouseEvent) => void;
  text: string;
}

const AppButton = ({
  buttonStyle,
  datae2eid,
  onClick,
  text
}: Props): ReactElement => (
  <StyledButton
    buttonStyle={buttonStyle}
    data-e2e-id={datae2eid}
    onClick={onClick}
    type="button"
  >
    {text}
  </StyledButton>
);

AppButton.propTypes = {
  buttonStyle: string,
  datae2eid: string,
  onClick: func,
  text: string.isRequired
};

AppButton.defaultProps = {
  buttonStyle: 'default',
  datae2eid: null,
  onClick: () => null
};

export default AppButton;
