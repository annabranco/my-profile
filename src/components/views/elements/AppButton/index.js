import React from 'react';
import { string, func } from 'prop-types';
import { StyledButton } from './styles';

const AppButton = ({ text, buttonStyle, onClick }) => (
  <StyledButton buttonStyle={buttonStyle} onClick={onClick} type="button">
    {text}
  </StyledButton>
);

AppButton.propTypes = {
  text: string.isRequired,
  buttonStyle: string,
  onClick: func
};

AppButton.defaultProps = {
  buttonStyle: 'default',
  onClick: () => null
};

export default AppButton;
