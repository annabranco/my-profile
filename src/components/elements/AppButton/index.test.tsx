import { shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';
import AppButton from '.';

// TODO [10-May -21]: Implement defined testing suite (Anna Branco)

describe('< AppButton >', () => {
  let wrapper: ShallowWrapper;

  beforeAll(() => {
    wrapper = shallow(
      <AppButton buttonStyle="" onClick={() => jest.fn()} text="Mock Button" />
    );
  });

  it('should mount', () => {
    expect(wrapper).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  xit('executes onClick prop event correctly', () => {
  });
});
