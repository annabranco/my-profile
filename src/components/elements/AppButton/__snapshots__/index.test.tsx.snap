// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< AppButton > matches the snapshot 1`] = `
<App Button
  buttonStyle=""
  data-e2e-id={null}
  onClick={[Function]}
  type="button"
>
  Mock Button
</App Button>
`;
